### ISSUE TITLE
Wrong type creation when using type: array

### ISSUE THREAD

**Issue by zaibon** [Jumpscale/go-raml#286]
Raml file: https://github.com/zaibon/rivine/blob/master/doc/api/api.raml

After generatic the python client  and trying to import it
Error : 
```
/home/zaibon/.virtualenv/lib/python3.6/rivine/Transaction.py in <module>()
      2 Auto-generated class for Transaction
      3 """
----> 4 from .array import array
      5 
      6 from . import client_support

ModuleNotFoundError: No module named 'rivine.array'
```

**Comment by iwanbk** [comment]
it is because the parser only doesn't recognise this as array
```
roottarget:
        type: array
        items: integer
```

it currently only support 
```
roottarget:
        type: integer[]
```

**Comment by iwanbk** [comment]
> it is because the parser only doesn't recognise this as array

which is ironic, because `typename[]` syntax is actually syntactic sugar.

**Comment by iwanbk** [comment]
This issue should be fixed by #289.
Corresponding test fixture has been added to https://github.com/Jumpscale/go-raml/blob/fix-array-type/codegen/fixtures/struct/struct.raml#L27
```
 colours:
        type: array
        items: string
```

But unfortunately your raml file still won't build properly because:
- it doesn't support this `body: Gateway` syntax. Should be fixed easily
- You use feature not supported by go-raml as addressed by #96, should have time next week to work on it.
